### ISSUE TITLE
JsonProperty annotation always uses a string / literal, better would be a constant

### ISSUE THREAD

**Issue by AnselmHa** [joelittlejohn/jsonschema2pojo#1088]
See **"sling:resourceType"** below is mentioned four times as literal. 
```
@JsonPropertyOrder({"sling:resourceType"})
public class Layout {
    @JsonProperty("sling:resourceType")
    private SlingResourceType slingResourceType;
    @JsonProperty("sling:resourceType")
    public SlingResourceType getSlingResourceType() { return slingResourceType; }
    @JsonProperty("sling:resourceType")
    public void setSlingResourceType(SlingResourceType slingResourceType) {
```
Better would be:
```
    public static final String SLING_RESOURCE_TYPE = "sling:resourceType";
    @JsonProperty(SLING_RESOURCE_TYPE)

```
As a positive side effect it would be possible to use the constant "SLING_RESOURCE_TYPE" from outside as well.

**Comment by ddcruver** [comment]
TL;DR Since I find no negative effects on maintainability I believe the property names should be left as literals for readability reasons.  But public static final constants for property names might be useful for usage outside of the generated objects.

Details:
I believe the reason why we use constants is because we want to prevent the accidental updating of it used in one place and not others.  Since this is generated code it is not prone to human errors such as these.

I believe the compiler would be smart enough reuse this string value effectively like a constant and abstracting the value in a constant would make it harder to ready

But I do agree there may be some value in adding the constants so they can reference the properties of the object without reflections. Changing of the property name would cause the constants to change causing a failure at compile-time for someone using the updated object property name constants. This would be good because if they used the property name as a literal themselves they would have to paying attention to the updated sachems and evaluating their code base and/or rely on their automated tests notifying them that the schema has name change properties.  To extend on this the pojo's generated from the schema could have a map of all properties.  I will leave it to others to voice if they believe this would be of benefit.

**Comment by joelittlejohn** [comment]
@ddcruver Yes, re why we don't use constants I'd agree. The current approach is more readable and simpler (both in this plugin and in the resulting code), and since the code is generated there's no obvious benefit where edits are concerned.

I'm not keen on adding extra generated code that we don't use or need. The naming is another thorny issue with no doubt edge cases and bugs to be raised. It can be easy to add new things to jsonschema2pojo but there's then an ongoing burden of support and maintenance, and every feature has to be brought along when we need to change something fundamental.

So I'm going to close this one. I can understand it might be useful, but we can't support everyone's idea about what kind of Java they would like.
